Event Object + Pagination

AliSzu opened the pull request.

AliSzu committed:
feat: event pagination

AliSzu committed:
test: event pagination

korzepadawid commented on backend/compose.yaml:
nit: you could name it `docker-compose.yaml`, specifying a filename while running `docker-compose up` will no longer be the case

korzepadawid replied:
other filenames are useful when you deal with multiple compose files

korzepadawid commented on backend/src/main/java/meetup/connect/Application.java:
```diff
@@ -10,4 +14,20 @@ public static void main(String[] args) {
+
+		private final DataGenerator dataGenerator;
+
+		@Autowired
```
nit: let's get rid of redundant `@Autowired` while injecting through a constructor

korzepadawid commented on backend/src/main/java/meetup/connect/Application.java:
```diff
@@ -10,4 +14,20 @@ public static void main(String[] args) {
 	}
 
+	@Component
+	public static class DataLoader implements CommandLineRunner {
```
suggestion: you could make `DataGenerator implements CommandLineRunner`

korzepadawid replied:
`dataGenerator.generateData()` won't be necessary then

korzepadawid commented on backend/src/main/java/meetup/connect/event/EventCreateDto.java:
```diff
@@ -0,0 +1,13 @@
+
+/**
+ * DTO for {@link Event}
+ */
```
nit: let's remove the redundant comment, the name of the class is meaningful

korzepadawid commented on backend/src/main/resources/application.properties:
```diff
@@ -1 +1,10 @@
+
+spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.properties.hibernate.format_sql= True
+spring.jpa.hibernate.ddl-auto=update
```
note: I kindly encourage you to delve deeper into `spring.jpa.hibernate.ddl-auto`

korzepadawid replied:
btw: looks good in this case(especially for the development purposes), however, you should be aware of how that works

korzepadawid commented on backend/src/test/java/meetup/connect/event/EventServiceTest.java:
```diff
@@ -0,0 +1,35 @@
+    @DisplayName("Should fetch events in paginated list")
+    void shouldFetchPaginatedEventsList() {
+
+        Page<Event> events = mock(Page.class);
```
suggestion: you could make the test more meaningful, by testing pagination on dummy data, explore https://www.baeldung.com/spring-data-jpa-convert-list-page

If you use some dummy data, you'll get 2 or more additional test cases:
- an empty page
- a not filled page
- a filled page with leftovers on the next page

korzepadawid commented on backend/src/test/java/meetup/connect/event/EventServiceTest.java:
```diff
@@ -0,0 +1,35 @@
+        Page<Event> events = mock(Page.class);
+        when(eventRepository.findAll(any(PageRequest.class))).thenReturn(events);
+
+        eventServiceImpl.getPage(1, 10);
```
note: Spring starts counting from 0 by default. Your code is fetching the second page. Was it deliberated?

korzepadawid replied:
I believe it should do the trick: https://stackoverflow.com/a/61603702 (verify pls)

korzepadawid commented on backend/src/main/java/meetup/connect/event/EventServiceImpl.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.List;
+
+@Service
+public class EventServiceImpl implements EventService {
```
suggestion: I don't think there's a need for a separate interface when you have only one implementation

korzepadawid commented on backend/src/main/java/meetup/connect/event/Event.java:
```diff
@@ -0,0 +1,50 @@
+@Entity
+@Getter
+@Setter
+public class Event {
```
praise: I'm a big fan of that file structure. It enables you to apply package-private access.

AliSzu committed:
suggested fixes

code reformat,.
more tests,
delete @Autowired for single construcotr,
change compose,yml name to docker-compose,.yml

AliSzu committed:
fix: delete EventServiceImpl
